src/coreclr/vm/frozenobjectheap.cpp

```diff
@@ -10,7 +10,8 @@
 #define FOH_COMMIT_SIZE (64 * 1024)
 
 FrozenObjectHeapManager::FrozenObjectHeapManager():
-    m_Crst(CrstFrozenObjectHeap, CRST_UNSAFE_COOPGC),
+    m_Crst(CrstFrozenObjectHeap, CRST_UNSAFE_ANYMODE),
+    m_SegmentRegistrationCrst(CrstFrozenObjectHeap, CRST_UNSAFE_ANYMODE),
```

Comment:
```suggestion
    m_SegmentRegistrationCrst(CrstFrozenObjectHeap),
```
This can be regular "safe" lock. It is always entered in preemptive mode.